api/src/services/__tests__/configService.ts

```diff
@@ -0,0 +1,57 @@
+import { existsSync, readFileSync } from 'fs';
+
+import { clearConfig, findConfig } from '../configService';
+
+jest.mock('fs');
+const mockExistsSync = jest.mocked(existsSync);
+const mockReadFileSync = jest.mocked(readFileSync);
+
+describe('configService', () => {
+  describe('findConfig', () => {
+    it('should read a secrets file if one exists with the key name', () => {
+      const key = 'TEST_KEY';
+      const secretValue = 'secret value';
+      mockExistsSync.mockReturnValue(true);
+      mockReadFileSync.mockReturnValue(secretValue);
+      expect(findConfig(key, 'default value')).toEqual(secretValue);
+      expect(mockExistsSync).toHaveBeenCalledWith(`/run/secrets/${key}`);
+      expect(mockReadFileSync).toHaveBeenCalledWith(
+        `/run/secrets/${key}`,
+        'utf-8'
+      );
+      clearConfig(key);
+    });
+
+    it("should read the config from the environment if a secrets file doesn't exist with the file name", () => {
+      const key = 'TEST_KEY';
+      process.env[key] = 'environment value';
+      mockExistsSync.mockReturnValue(false);
+      expect(findConfig(key, 'default value')).toEqual(process.env[key]);
+      expect(mockExistsSync).toHaveBeenCalledWith(`/run/secrets/${key}`);
+      expect(mockReadFileSync).not.toHaveBeenCalled();
+      delete process.env[key];
+      clearConfig(key);
+    });
+
+    it('should use the default value if the config is nether a secret nor in the environment', () => {
+      const key = 'TEST_KEY';
+      const defaultValue = 'default value';
+      mockExistsSync.mockReturnValue(false);
```

Comment:
The test implies that the config isn't in the environment, but should it be more explicit (i.e. would it be valuable to set an environment variable at the `key` to a `null` value or `undefined`)?

Reply:
The `process.env` global will never have a property set to `null` or `undefined` in normal circumstances. Either it will have an "own" property set to a string, or it won't. In the case where this test fails because the environment indeed has a non-empty string value for the test key, then it would be an exceptional case and this test would fail. That would be a good thing because it would reveal the exception instead of silently passing.